### ISSUE TITLE
SBT global directory should be chosen by binary version

### ISSUE THREAD

**Issue by jsuereth** [sbt/sbt#735]
Because of the way the world works, it is very rare to have the same version of a plugin published against multiple SBT versions.   WE also no longer publish sbt plugins as source w/ hefty source compatibility.   This leads to frustrations with the ~/.sbt/ and ~/.sbt/plugins/ directory.

We'd like to specify plugins only once and re-use them as long as they are compatible with existing sbt versions.  THe simplest approach is taken by sbt "rebel" cut, where the rebellion  simply appends the sbt binary version after the "~/.sbt" directory and declares this "boot".

Given that this technique has seen proven happy users, we should make it the default for the sbt launcher, and adjust documentation appropriately for 0.13.   

Keep rawking it.

**Comment by harrah** [comment]
The only interesting part in the commit is the transition warning.  If you have global configuration in the previous default location (`~/.sbt/`) but not in the new location, there is a warning that looks like:

```
[warn] The global sbt directory is now versioned and is located at /home/user/.sbt/0.13.
[warn]   You are seeing this warning because there is global configuration in /user/.sbt but not in /home/user/.sbt/0.13.
[warn]   The global sbt directory may be changed via the sbt.global.base system property.
```

Configuration is either `.sbt` files or a `plugins` directory.  I'm open to alternatives, but some kind of warning is needed so that users don't wonder why their global settings aren't working.  This warning only applies when the defaults are used and _not_ when something like sbt-extras explicitly sets the global directory.

**Comment by jsuereth** [comment]
Just a note:  We should warn the sbt-extras users, because this may break
them if they specify an "Sbt global" environment variable....

**Comment by harrah** [comment]
This does not override explicit configuration.

**Comment by jsuereth** [comment]
Oh?  I thought the default were expanded in the launcher.

**Comment by harrah** [comment]
Not sure what you mean.  `getFileSetting` checks if a system property is set and if not uses the default.

**Comment by jsuereth** [comment]
Ah, gotcha.  I'm thinking of different global settings in the launcher, which the script also overrides.   So we should have similar sbt loading behavior, but we may still have different boot directories, e.g.   That seems... fine, right?

**Comment by adriaanm** [comment]
when you use the `0.13.+` notation in your `build.properties`, you get this suspicious-looking warning -- shouldn't the + be resolved to the current version?

```
[warn] The global sbt directory is now versioned and is located at /Users/adriaan/.sbt/0.13.+.
[warn]   You are seeing this warning because there is global configuration in /Users/adriaan/.sbt but not in /Users/adriaan/.sbt/0.13.+.
[warn]   The global sbt directory may be changed via the sbt.global.base system property.
```

**Comment by jsuereth** [comment]
I believe this might be an issue with the binary version detection.

**Comment by adriaanm** [comment]
It also seems to break plugin resolution, looking for a `sbt_0.13.+` subdirectory:

```
[warn] ==== typesafe-ivy-releases: tried
[warn]   http://repo.typesafe.com/typesafe/ivy-releases/com.typesafe.sbteclipse/sbteclipse-plugin/scala_2.10/sbt_0.13.+/2.2.0/ivys/ivy.xml
[warn] ==== sbt-plugin-releases: tried

```

**Comment by harrah** [comment]
The `0.13.+` problem is a separate issue from binary version detection and the global directory.  Right now, by sort-of-design, the launcher only supports stable versions and -SNAPSHOT.  When the launcher resolves the application, it would need to pull out the version that was resolved, persist it for future runs (not currently done anywhere by the launcher), pass that to the launched application, and the launched application would need to use that resolved version instead of the requested version.  So, please open a separate enhancement for this.
